lib/cocina/models.rb

```diff
@@ -25,8 +25,23 @@ def camelize(basename, _abspath)
 loader.setup
 
 module Cocina
+  # Provides Ruby objects for the repository and serializing them to/from JSON.
   module Models
     class Error < StandardError; end
-    # Your code goes here...
+
+    # @param [Hash] dyn a ruby hash representation of the JSON serialization of a collection or DRO
+    # @return [DRO,Collection]
+    # @raises [RuntimeError] if a valid type is not found in the data
+    # @raises [KeyError] if a type field cannot be found in the data
+    def self.build(dyn)
+      case dyn.fetch('type')
+      when *DRO::TYPES
+        DRO.from_dynamic(dyn)
+      when *Collection::TYPES
+        Collection.from_dynamic(dyn)
+      else
+        raise "Unknown type: '#{dyn.fetch('type')}'"
```

Comment:
Worth raising the custom error type defined on line 30 here so clients get a more specific error? Or creating a new error subtype for unknown types?

Reply:
Yes, fixed.